onmt/decoders/transformer.py

```diff
@@ -717,8 +717,7 @@ def _forward(
         dec_mask = None
 
         if layer_in.size(1) > 1:
-            # masking is necessary when sequence length is greater than one
-            dec_mask = self._compute_dec_mask(tgt_pad_mask, future)
+            dec_mask = tgt_pad_mask
```

Comment:
you cannot simply remove the triu mask. I think the easiest thing to do for now is:
if step is None:
 dec_mask = self._compute_de_mask(tgt_pad_mask, future) # training case
else:
 dec_mask = tgt_pad_mask

Reply:
if step is _not_ None ?

Reply:
step=None => training we need the triu mask
step=0 or any int => Inference. In theory for step 0 we should need the mask but not in the case of left padding

Reply:
for reference we reverted back to original code but patched the dec_mask in _compute_dec_mask because of this bug: https://github.com/pytorch/pytorch/issues/103749
hopefully will be resolved later on.